src/main/java/io/vertx/core/impl/launcher/VertxLifecycleHooks.java

```diff
@@ -0,0 +1,61 @@
+/*
+ *  Copyright (c) 2011-2015 The original author or authors
+ *  ------------------------------------------------------
+ *  All rights reserved. This program and the accompanying materials
+ *  are made available under the terms of the Eclipse Public License v1.0
+ *  and Apache License v2.0 which accompanies this distribution.
+ *
+ *       The Eclipse Public License is available at
+ *       http://www.eclipse.org/legal/epl-v10.html
+ *
+ *       The Apache License v2.0 is available at
+ *       http://www.opensource.org/licenses/apache2.0.php
+ *
+ *  You may elect to redistribute this code under either of these licenses.
+ */
+package io.vertx.core.impl.launcher;
+
+import io.vertx.core.DeploymentOptions;
+import io.vertx.core.Vertx;
+import io.vertx.core.VertxOptions;
+
+/**
+ * Interface that let sub-classes of launcher to be notified on different events.
+ *
+ * @author Clement Escoffier <clement@apache.org>
+ */
+public interface VertxLifecycleHooks {
+
+  /**
+   * Hook for sub-classes of the starter class before the vertx instance is started.
```

Comment:
you should perhaps mention if it is safe or not to mutate the options in the doc.